### PR TITLE
refactor: `CircuitVariable`

### PR DESCRIPTION
**Pull request by puma314**
## Overview
- Refactor `CircuitVariable` to remove previously slow `elements`, `from_elements` and `nb_elements` functions. 
- `CircuitVariable` implements `get`, `set` and `constant` in the trait. 
- Implementers of `CircuitVariable` will need to define `nb_elements`, `from_elements` and `elements`, or derive them from existing `CircuitVariable` implmentations.

### CODE REVIEW

File: plonky2x/src/frontend/eth/vars.rs [hunk 1]
```diff
@@ -23,11 +22,16 @@ impl CircuitVariable for BLSPubkeyVariable {
         Self(BytesVariable::init_unsafe(builder))
     }
 
-    fn constant<L: PlonkParameters<D>, const D: usize>(
-        builder: &mut CircuitBuilder<L, D>,
-        value: Self::ValueType<L::Field>,
-    ) -> Self {
-        Self(BytesVariable::constant(builder, value))
+    fn nb_elements() -> usize {
```
Comment (ratankaliani) [inline plonky2x/src/frontend/eth/vars.rs:25]: nit: Is there a better way to do this?

File: plonky2x/src/frontend/eth/vars.rs [hunk 2]
```diff
@@ -66,14 +62,16 @@ impl CircuitVariable for AddressVariable {
...
 
-    fn constant<L: PlonkParameters<D>, const D: usize>(
-        builder: &mut CircuitBuilder<L, D>,
-        value: Self::ValueType<L::Field>,
-    ) -> Self {
-        Self(BytesVariable::constant(
-            builder,
-            value.as_bytes().try_into().expect("wrong slice length"),
-        ))
+    fn nb_elements() -> usize {
```
Comment (ratankaliani) [inline plonky2x/src/frontend/eth/vars.rs:65]: nit: Is there a better way to do this?

File: plonky2x/src/frontend/eth/vars.rs (hunk 1 above)
Comment (puma314) [reply plonky2x/src/frontend/eth/vars.rs:25]: I don't think so, this is what I also do

**Review by ratankaliani — APPROVED** [review]
lgtm!

**Pull request merged**